packages/list/lib/index.ts

```diff
@@ -1,20 +1,21 @@
 #!/usr/bin/env node
 
-const { autoDetect } = require('@serialport/bindings-cpp')
+import { autoDetect, PortInfo } from '@serialport/bindings-cpp'
+import { program, Option } from 'commander'
+
 const { version } = require('../package.json')
-const { program, Option } = require('commander')
 
 const formatOption = new Option('-f, --format <type>', 'Format the output').choices(['text', 'json', 'jsonline', 'jsonl']).default('text')
 
 program.version(version).description('List available serial ports').addOption(formatOption).parse(process.argv)
 
-function jsonl(ports) {
+function jsonl(ports: PortInfo[]) {
   ports.forEach(port => {
     console.log(JSON.stringify(port))
   })
 }
 
-const formatters = {
+const formatters: { [key: string]: undefined | ((ports: PortInfo[]) => void) } = {
```

Comment:
I find the Record type helpful to break up the "delimiter soup" of object types, ideally making it clearer that `formatters` is an object with any number of string keys which have a value of undefined or a `ports` callback. 

```suggestion
const formatters: Record<string, undefined | ((ports: PortInfo[]) => void)> = {
```